src/app/components/grid/firesides-channel.ts

```diff
@@ -0,0 +1,36 @@
+import { shallowReadonly } from 'vue';
+import { createLogger } from '../../../utils/logging';
+import { createSocketChannelController } from '../../../_common/socket/socket-controller';
+import { GridClient } from './client.service';
+
+export type GridFiresidesChannel = ReturnType<typeof createGridFiresidesChannel>;
+
+interface FiresideExpiredPayload {
+	fireside_hash: string;
+}
+
+export function createGridFiresidesChannel(client: GridClient) {
+	const { socketController, appStore } = client;
+
+	const logger = createLogger('Fireside');
+
+	const channelController = createSocketChannelController('firesides', socketController);
+
+	channelController.listenTo('expired', _onFiresideExpired);
+
+	const joinPromise = channelController.join({
+		async onJoin() {},
```

Comment:
Do we need to do anything during `onJoin`? We can just remove that object if we don't need to do anything when joining.